packages/back-end/src/controllers/experiments.ts

```diff
@@ -1011,6 +1012,16 @@ export async function postExperiment(
     }
   }
 
+  if (data.variationWeights) {
+    const phases = [...experiment.phases];
+    const lastIndex = phases.length - 1;
+    phases[lastIndex] = {
+      ...phases[lastIndex],
+      variationWeights: data.variationWeights,
+    };
+    changes.phases = phases;
+  }
```

Comment:
Should we do any validation in the controller to prevent this from happening to running experiments or something like that? Feels like this exposes us to some risk if someone just starts using `postExperiment` without knowing they could be affecting the prod weights? But maybe that's just the price we have to pay.

Reply:
I think its the price we pay for now at least. We have lots of these partial edit widgets that use the same endpoint.

Reply:
Yeah, this just seems like a critical one that affects the actual `feature`, which in general we should strive to put more safety around, rather than less. It's already confusing when you make an experiment change and it auto-affects a feature people thought was approval flow controlled.

Reply:
i think we should put effort into locking the controller logic (not just for this widget). i can take a look at that in a follow up pr